benchmarks/performance_test.py

```diff
@@ -1,17 +1,78 @@
 from urlparse4 import urlsplit, urljoin
 from timeit import default_timer as timer
 
-total = 0
+from sys import argv
 
-with open('urls/chromiumUrls.txt') as f:
-    for url in f:
+encode = False
 
-        start = timer()
+try:
+    if argv[1] == "encode":
+        encode = True
+except IndexError:
+    print("encode is not defined, continue with the program...")
 
-        a = urlsplit(url)
+if encode:
+    total = 0
 
-        end = timer()
+    for i in range(5):
+        with open('urls/chromiumUrls.txt') as f:
+            for url in f:
 
-        total += end - start
+                start = timer()
 
-print("the total time is", total, "seconds")
+                a = urlsplit(url.encode())
+
+                end = timer()
+
+                total += end - start
+
+    print("the urlsplit time with encode in python is", total / 5, "seconds")
+
+
+    total2 = 0
+
+    for i in range(5):
+        with open('urls/chromiumUrls.txt') as f:
+            for url in f:
+
+                start = timer()
+
+                a = urljoin(url.encode(), "/asd".encode())
```

Comment:
I think it's better to move encoding of url and path outside of timed part, and specify encoding

Reply:
Also this will allow to remove repetition of code for when encode is True or False - instead of repeating whole bodies, it would be possible to have just a small condition